esmf_regrid/_esmf_sdo.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""Provides representations of ESMF's Spatial Discretisation Objects."""
+
+from abc import ABC, abstractmethod
+
+import cartopy.crs as ccrs
+import ESMF
+import numpy as np
+
+
+class SDO(ABC):
+    """
+    Abstract base class for handling spatial discretisation objects.
+
+    This contains shared things for representing the three spatial discretisation
+    objects supported by ESMPy, Grids, Meshes, and LocStreams.
+    """
+
+    def __init__(self, shape, index_offset, field_kwargs):
+        self._shape = shape
+        self._index_offset = index_offset
+        self._field_kwargs = field_kwargs
+
+    @abstractmethod
+    def _make_esmf_sdo(self):
+        pass
+
+    def make_esmf_field(self):
+        """Return an ESMF field representing the grid."""
```

Comment:
References to "grid" ought to be replaced, perhaps with either "object" or "spatial discretisation object".